Cannot install daemonupgradecontroller via readme

I'm try to install ds controller via manual
https://github.com/Mirantis/k8s-daemonupgradecontroller/blob/master/README.md

vagrant@node1:~/k8s-daemonsetupgradecontroller$ make build
go get github.com/Masterminds/glide
package github.com/Masterminds/glide: cannot download, $GOPATH not set. For more details see: go help gopath
Makefile:91: recipe for target 'vendor' failed
make: *** [vendor] Error 1

So please update readme and provide full info for installing

also next issues 

vagrant@node1:~/golang/src/github.com/Mirantis/k8s-daemonupgradecontroller$ make build
mkdir -p _output
go get github.com/Masterminds/glide
glide install --strip-vendor
[INFO]	Downloading dependencies. Please wait...
[INFO]	--> Fetching github.com/go-openapi/jsonpointer.
[INFO]	--> Fetching github.com/dgrijalva/jwt-go.
[INFO]	--> Fetching github.com/go-openapi/jsonreference.
[INFO]	--> Fetching bitbucket.org/ww/goautoneg.
[INFO]	--> Fetching cloud.google.com/go.
[INFO]	--> Fetching github.com/Azure/go-ansiterm.
[INFO]	--> Fetching github.com/beorn7/perks.
[INFO]	--> Fetching github.com/blang/semver.
[INFO]	--> Fetching github.com/coreos/go-oidc.
[INFO]	--> Fetching github.com/coreos/pkg.
[INFO]	--> Fetching github.com/davecgh/go-spew.
[INFO]	--> Fetching github.com/docker/go-units.
[INFO]	--> Fetching github.com/docker/distribution.
[INFO]	--> Fetching github.com/docker/docker.
[INFO]	--> Fetching github.com/docker/engine-api.
[INFO]	--> Fetching github.com/docker/go-connections.
[INFO]	--> Fetching github.com/emicklei/go-restful.
[INFO]	--> Fetching github.com/docker/spdystream.
[INFO]	--> Fetching github.com/exponent-io/jsonpath.
[INFO]	--> Fetching github.com/ghodss/yaml.
[WARN]	Unable to checkout bitbucket.org/ww/goautoneg
[ERROR]	Update failed for bitbucket.org/ww/goautoneg: hg is not installed
[INFO]	--> Fetching github.com/go-openapi/spec.
[INFO]	--> Fetching github.com/go-openapi/swag.
[INFO]	--> Fetching github.com/gogo/protobuf.
[INFO]	--> Fetching github.com/golang/glog.
[INFO]	--> Fetching github.com/golang/groupcache.
[INFO]	--> Fetching github.com/golang/protobuf.
[INFO]	--> Fetching github.com/google/gofuzz.
[INFO]	--> Fetching github.com/howeyc/gopass.
[INFO]	--> Fetching github.com/imdario/mergo.
[INFO]	--> Fetching github.com/jonboulle/clockwork.
[INFO]	--> Fetching github.com/juju/ratelimit.
[INFO]	--> Fetching github.com/mailru/easyjson.
[INFO]	--> Fetching github.com/matttproud/golang_protobuf_extensions.
[INFO]	--> Fetching github.com/mitchellh/go-wordwrap.
[INFO]	--> Fetching github.com/onsi/ginkgo.
[INFO]	--> Fetching github.com/onsi/gomega.
[INFO]	--> Fetching github.com/pborman/uuid.
[INFO]	--> Fetching github.com/prometheus/client_golang.
[INFO]	--> Fetching github.com/prometheus/client_model.
[INFO]	--> Fetching github.com/prometheus/common.
[INFO]	--> Fetching github.com/prometheus/procfs.
[INFO]	--> Fetching github.com/PuerkitoBio/purell.
[INFO]	--> Fetching github.com/PuerkitoBio/urlesc.
[INFO]	--> Fetching github.com/Sirupsen/logrus.
[INFO]	--> Fetching github.com/spf13/pflag.
[INFO]	--> Fetching github.com/ugorji/go.
[INFO]	--> Fetching golang.org/x/crypto.
[INFO]	--> Fetching golang.org/x/net.
[INFO]	--> Fetching golang.org/x/oauth2.
[INFO]	--> Fetching golang.org/x/sys.
[INFO]	--> Fetching golang.org/x/text.
[INFO]	--> Fetching google.golang.org/appengine.
[INFO]	--> Fetching gopkg.in/inf.v0.
[INFO]	--> Fetching gopkg.in/yaml.v2.
[INFO]	--> Fetching k8s.io/kubernetes.
[ERROR]	Failed to install: hg is not installed
Makefile:91: recipe for target 'vendor' failed
make: *** [vendor] Error 1

--- a/README.md
+++ b/README.md
@@ -1,35 +1,128 @@
 [![Build Status](https://travis-ci.org/Mirantis/k8s-daemonupgradecontroller.svg?branch=master)](https://travis-ci.org/Mirantis/k8s-daemonupgradecontroller)
 Daemon Set upgrade controller
-============================
+=============================
 
-Download
-```
-https://github.com/Mirantis/k8s-daemonsetupgradecontroller
+Controller which manages Kubernetes DaemonSet upgrades. It stores history using PodTemplates,
+allows rollback to previous version and pause the upgrade process.
+
+Getting Started
+===============
+
+Installing
+----------
+
+Daemon upgrade controller can be started as a docker container inside of the Kubernetes cluster.
+
+Create new file called `controller.yml`
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: daemonupgrader
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: daemonupgrader
+    spec:
+      containers:
+      - name: daemonupgrader
+        image: mirantis/k8s-daemonupgradecontroller:latest
+        imagePullPolicy: IfNotPresent
 ```
 
-Build
-```
-make build
-```
-
-Build docker image
-```
-make docker
-```
+and start the container: `kubectl --namespace kube-system -f controller.yml`
 
 Usage
-=====
+-----
+
+All configuration is done by setting options in daemon annotations.
+
+By default upgrade controller will not touch the running daemons.
+To enable upgrades, strategy type needs to be set:
+
+`daemonset.kubernetes.io/strategyType: "RollingUpdate"`
+
+from now on whenever `Spec.Template` is changed upgrade controller will delete old pods
+and new pods with new template will be created. By default controller deletes one pod at the time and
+waits for new to be in Ready state. It can by configured by setting how many pods can be unavailable 
+during the upgrade:
+
+`daemonset.kubernetes.io/maxUnavailable: "1"`
 
 [![asciicast](https://asciinema.org/a/94027.png)](https://asciinema.org/a/94027)
 
-How to run unit tests?
-======================================
-Unit tests:
+All configuration options
+-------------------------
+
+#### daemonset.kubernetes.io/paused
+
+When set to "true" or "1" it pauses upgrades.
+
+#### daemonset.kubernetes.io/rollbackTo
+
+Sets the revision number to which rollback. When rollback is done it's set to empty string.
+
+#### daemonset.kubernetes.io/strategyType
+
+For now two strategies are supported:
+
+* RollingUpdate - deletes old pods and waiting for new to be created. By default it deletes one pod at the time.
+* Noop - does nothing.
+
+#### daemonset.kubernetes.io/maxUnavailable
+
+Sets how many pods can be deleted in one step during the upgrade. Default: 1.
+
+Building from source
+====================
+
+#### Rquirements
+
+* make
+* golang
+* docker
+* git
+* mercurial
+* configured go environemnt
+
+Put this project in the $GOPATH
+
+#### Building
+
+```shell
+make build
+```
+
+##### docker image
+
+```shell
+make build-image
+```
+
+#### Running unit tests
+
 ```
 make unit
 ```
 
-How to run e2e tests?
-=====================
-TODO
+#### Running e2e tests
 
+Start container with daemon upgrade controller. Make sure that it can connect to Kubernetes cluster. If [kubernetes dind cluster](https://github.com/sttts/kubernetes-dind-cluster) is used
+then container can be started using [docker-compos.yaml](https://github.com/Mirantis/k8s-daemonupgradecontroller/blob/master/docker-compose.yaml) file.
+
+```shell
+make e2e
+```
+
+Authors
+=======
+
+Łukasz Oleś, github: lukaszo, IRC: salmon_, k8s slack: lukaszo
+
+License
+=======
+
+Apache 2.0 License

--- a/deployment.yaml
+++ b/deployment.yaml
@@ -11,5 +11,5 @@ spec:
     spec:
       containers:
       - name: daemonupgrader
-        image: mirantis/k8s-daemonupgradecontroller:0.1
+        image: mirantis/k8s-daemonupgradecontroller:latest
         imagePullPolicy: IfNotPresent

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -1,6 +1,6 @@
 daemonupgradecontroller:
   hostname: daemonupgradecontroller
-  image: mirantis/k8s-daemonupgradecontroller:0.1
+  image: mirantis/k8s-daemonupgradecontroller
   command:
   - --master=http://dind_apiserver_1:8888
   - --v=4

